diff --git a/pristine/squid-7.5/src/base/File.cc b/patched/squid-7.5/src/base/File.cc
--- a/pristine/squid-7.5/src/base/File.cc
+++ b/patched/squid-7.5/src/base/File.cc
@@ -215,9 +215,10 @@
 File::truncate()
 {
 #if _SQUID_WINDOWS_ || _SQUID_MINGW_
-    if (!SetFilePointer(fd_, 0, nullptr, FILE_BEGIN)) {
-        const auto savedError = GetLastError();
-        throw TexcHere(sysCallFailure("SetFilePointer", WindowsErrorMessage(savedError)));
+    LARGE_INTEGER truncateOffset = {};
+    if (!SetFilePointerEx(fd_, truncateOffset, nullptr, FILE_BEGIN)) {
+        const auto savedError = GetLastError();
+        throw TexcHere(sysCallFailure("SetFilePointerEx", WindowsErrorMessage(savedError)));
     }
 
     if (!SetEndOfFile(fd_)) {
