diff --git a/pristine/squid-7.5/configure b/patched/squid-7.5/configure
index 16c1790..2b8fbd5 100644
--- a/pristine/squid-7.5/configure
+++ b/patched/squid-7.5/configure
@@ -6634,6 +6634,10 @@ then :
   enableval=$enable_dependency_tracking;
 fi
 
+# squid4win-mingw-disable-deptrack
+case $host_os in
+  mingw*) enable_dependency_tracking=no ;;
+esac
 if test "x$enable_dependency_tracking" != xno; then
   am_depcomp="$ac_aux_dir/depcomp"
   AMDEPBACKSLASH='\'
@@ -23007,7 +23011,7 @@ then :
 
 fi
 
-    MINGW_LIBS="-lmingwex"
+    MINGW_LIBS="-lmingwex -lws2_32"
 
     ac_fn_cxx_check_func "$LINENO" "fsync" "ac_cv_func_fsync"
 if test "x$ac_cv_func_fsync" = xyes
@@ -26785,6 +26789,10 @@ fi
 printf "%s\n" "$ac_cv_require_wno_deprecated_register" >&6; }
 
 
+# squid4win-mingw-disable-strict-errors
+case $host_os in
+  mingw*) enable_strict_error_checking=no ;;
+esac
 if test "x$enable_strict_error_checking" != "xno"
 then :
 
diff --git a/pristine/squid-7.5/configure.ac b/patched/squid-7.5/configure.ac
index 3643515..1588482 100644
--- a/pristine/squid-7.5/configure.ac
+++ b/patched/squid-7.5/configure.ac
@@ -219,7 +219,7 @@ AS_CASE(["$squid_host_os"],
 #include <windows.h>
 #endif
     ])
-    MINGW_LIBS="-lmingwex"
+    MINGW_LIBS="-lmingwex -lws2_32"
     AC_SUBST(MINGW_LIBS)
     AC_CHECK_FUNCS([fsync])
   ],
diff --git a/pristine/squid-7.5/libltdl/configure b/patched/squid-7.5/libltdl/configure
index 2bf4d68..223ed78 100644
--- a/pristine/squid-7.5/libltdl/configure
+++ b/patched/squid-7.5/libltdl/configure
@@ -3545,6 +3545,10 @@ then :
   enableval=$enable_dependency_tracking;
 fi
 
+# squid4win-mingw-disable-deptrack
+case $host_os in
+  mingw*) enable_dependency_tracking=no ;;
+esac
 if test "x$enable_dependency_tracking" != xno; then
   am_depcomp="$ac_aux_dir/depcomp"
   AMDEPBACKSLASH='\'
diff --git a/pristine/squid-7.5/src/auth/basic/RADIUS/Makefile.am b/patched/squid-7.5/src/auth/basic/RADIUS/Makefile.am
index c98169a..82d1d01 100644
--- a/pristine/squid-7.5/src/auth/basic/RADIUS/Makefile.am
+++ b/patched/squid-7.5/src/auth/basic/RADIUS/Makefile.am
@@ -23,4 +23,5 @@ basic_radius_auth_LDADD= \
 	$(COMPAT_LIB) \
 	$(LIBNETTLE_LIBS) \
 	$(SSLLIB) \
-	$(XTRA_LIBS)
+	$(XTRA_LIBS) \
+	$(MINGW_LIBS)
diff --git a/pristine/squid-7.5/src/auth/basic/RADIUS/Makefile.in b/patched/squid-7.5/src/auth/basic/RADIUS/Makefile.in
index ef8e561..e494dc9 100644
--- a/pristine/squid-7.5/src/auth/basic/RADIUS/Makefile.in
+++ b/patched/squid-7.5/src/auth/basic/RADIUS/Makefile.in
@@ -760,7 +760,8 @@ basic_radius_auth_LDADD = \
 	$(COMPAT_LIB) \
 	$(LIBNETTLE_LIBS) \
 	$(SSLLIB) \
-	$(XTRA_LIBS)
+	$(XTRA_LIBS) \
+	$(MINGW_LIBS)
 
 all: all-am
 
diff --git a/pristine/squid-7.5/src/security/cert_generators/file/Makefile.am b/patched/squid-7.5/src/security/cert_generators/file/Makefile.am
index 6668018..978ebdc 100644
--- a/pristine/squid-7.5/src/security/cert_generators/file/Makefile.am
+++ b/patched/squid-7.5/src/security/cert_generators/file/Makefile.am
@@ -34,5 +34,6 @@ security_file_certgen_LDADD = \
 	$(top_builddir)/src/base/libbase.la \
 	$(top_builddir)/src/time/libtime.la \
 	$(SSLLIB) \
+	$(MINGW_LIBS) \
 	$(COMPAT_LIB)
 
diff --git a/pristine/squid-7.5/src/security/cert_generators/file/Makefile.in b/patched/squid-7.5/src/security/cert_generators/file/Makefile.in
index d146174..b50fc67 100644
--- a/pristine/squid-7.5/src/security/cert_generators/file/Makefile.in
+++ b/patched/squid-7.5/src/security/cert_generators/file/Makefile.in
@@ -782,6 +782,7 @@ security_file_certgen_LDADD = \
 	$(top_builddir)/src/base/libbase.la \
 	$(top_builddir)/src/time/libtime.la \
 	$(SSLLIB) \
+	$(MINGW_LIBS) \
 	$(COMPAT_LIB)
 
 all: all-am
