diff --git a/pristine/squid-7.5/src/fs_io.cc b/patched/squid-7.5/src/fs_io.cc
--- a/pristine/squid-7.5/src/fs_io.cc
+++ b/patched/squid-7.5/src/fs_io.cc
@@ -450,7 +450,7 @@
     // ensure c_str() lifetimes even if `to` and `from` share memory
     SBuf to2(to.rawContent(), to.length());
 
-#if _SQUID_OS2_ || _SQUID_WINDOWS_
+#if _SQUID_OS2_ || _SQUID_WINDOWS_ || _SQUID_MINGW_
     remove(to2.c_str());
 #endif
 
