diff --git a/pristine/squid-7.5/src/helper.cc b/patched/squid-7.5/src/helper.cc
--- a/pristine/squid-7.5/src/helper.cc
+++ b/patched/squid-7.5/src/helper.cc
@@ -72,7 +72,7 @@
 void
 Helper::SessionBase::closePipesSafely()
 {
-#if _SQUID_WINDOWS_
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
     shutdown(writePipe->fd, SD_BOTH);
 #endif
 
@@ -83,7 +83,7 @@
         readPipe->close();
     writePipe->close();
 
-#if _SQUID_WINDOWS_
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
     if (hIpc) {
         if (WaitForSingleObject(hIpc, 5000) != WAIT_OBJECT_0) {
             getCurrentTime();
@@ -98,7 +98,7 @@
 void
 Helper::SessionBase::closeWritePipeSafely()
 {
-#if _SQUID_WINDOWS_
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
     shutdown(writePipe->fd, (readPipe->fd == writePipe->fd ? SD_BOTH : SD_SEND));
 #endif
 
@@ -107,7 +107,7 @@
         readPipe->fd = -1;
     writePipe->close();
 
-#if _SQUID_WINDOWS_
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
     if (hIpc) {
         if (WaitForSingleObject(hIpc, 5000) != WAIT_OBJECT_0) {
             getCurrentTime();
